Add options button to main menu

tpenguinltg opened the pull request:
Adding this button to the main menu allows a new user (i.e. one who has no beatmaps) to access the options without installing a beatmap.
We may also want to consider making a new image so that it doesn't say "Other Options" when there are no other options on that screen.

Fixes #279

yugecin commented on src/itdelatrisu/opsu/states/MainMenu.java:
```diff
@@ -720,6 +783,15 @@ public void mousePressed(int button, int x, int y) {
 		}
 
+		// options button actions
+		if (selectOptionsButton.contains(x, y)) {
```
put this in `mouseClicked` to solve the disappearing problem.
You'll also need to add 
```
if (showOptionsOverlay || !optionsOverlayProgress.isFinished()) return;
```
at the top of `mouseClicked`

tpenguinltg replied:
Looks like that fixed it. Thanks.

yugecin commented:
The CTRL+O shortcut should be added aswell (like in SongMenu)

tpenguinltg commented:
@yugecin: Thanks for reviewing. Would it matter if it's just O instead of Ctrl+O? I think the only reason it's Ctrl+O in SongMenu is because O would conflict with search.

tpenguinltg commented:
I just noticed that searching is broken. The search field seems to lose focus every time a key is pressed, and it doesn't even get focus when using the keyboard shortcut to open the overlay. Any ideas?

yugecin commented:
Something weird is happening with the events. Seems like the search textfield consumes the events in these cases, but not when the options menu is opened via the song menu.
No idea why it does that, but I think I found a workaround/fix that seems to be working by playing with the focus of the field:

Remove line https://github.com/itdelatrisu/opsu/blob/8dd92de527213c910141d43c0be5ac152247df5e/src/itdelatrisu/opsu/options/OptionsOverlay.java#L368
and add `searchField.setFocus(false);` after the key press was sent: https://github.com/itdelatrisu/opsu/blob/8dd92de527213c910141d43c0be5ac152247df5e/src/itdelatrisu/opsu/options/OptionsOverlay.java#L1070

I propose CTRL+O because it's also how it's done in osu!, and it makes it consistent with the keybind in the song menu (indeed to prevent conflict with search).
In osu! the O key is also used to open the main menu (like when you click the logo), so that would cause conflicts aswell if that would be added to opsu!.

tpenguinltg commented:
Weird, but it seems to work. Nice job finding that.

I implemented that fix and changed the keybinding to Ctrl+O. I don't see any other issues with this, so unless someone else spots something, I think this is good to merge.

itdelatrisu commented:
Thank you!  Sorry for the late response.  I've committed my own patch (f4fa217) largely based on this, with these changes:
* Didn't like having the image there, so this is keyboard-only for now.
* Used the full osu! behavior for `O` vs. `CTRL+O` as @yugecin described above.
* Moved a bunch of things around.
